tasks.c

```diff
@@ -6897,13 +6974,44 @@ static void prvResetNextTaskUnblockTime( void )
+                    pcWriteBuffer = prvWriteNameToBuffer( pcWriteBuffer, pxTaskStatusArray[ x ].pcTaskName );
+                    uxConsumedBufferLength += configMAX_TASK_NAME_LEN;
+                }
+                else
+                {
+                    /* Output buffer full. */
+                    break;
+                }
+
+                /* Is there space left in the buffer? -1 is done because snprintf
+                 * writes a terminating null character. So we are essentially
+                 * checking if the buffer has space to write at least one non-null
+                 * character. */
+                if( uxConsumedBufferLength < ( uxBufferLength - 1 ) )
+                {
+                    /* Write the rest of the string. */
+                    lSnprintfReturnValue = snprintf( pcWriteBuffer,
+                                                     uxBufferLength - uxConsumedBufferLength,
+                                                     "\t%c\t%u\t%u\t%u\r\n",
+                                                     cStatus,
+                                                     ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority,
+                                                     ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark,
+                                                     ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber ); /*lint !e586 sprintf() allowed as this is compiled with many compilers and this is a utility function only - not part of the core kernel implementation. */
+                    uxCharsWrittenBySnprintf = prvSnprintfReturnValueToCharsWritten( lSnprintfReturnValue, uxBufferLength - uxConsumedBufferLength );
 
-                /* Write the rest of the string. */
-                sprintf( pcWriteBuffer, "\t%c\t%u\t%u\t%u\r\n", cStatus, ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority, ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark, ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber ); /*lint !e586 sprintf() allowed as this is compiled with many compilers and this is a utility function only - not part of the core kernel implementation. */
-                pcWriteBuffer += strlen( pcWriteBuffer );                                                                                                                                                                                                /*lint !e9016 Pointer arithmetic ok on char pointers especially as in this case where it best denotes the intent of the code. */
+                    uxConsumedBufferLength += uxCharsWrittenBySnprintf;
+                    pcWriteBuffer += uxCharsWrittenBySnprintf; /*lint !e9016 Pointer arithmetic ok on char pointers especially as in this case where it best denotes the intent of the code. */
+                }
+                else
+                {
+                    /* Output buffer full. */
+                    break;
```

Comment:
MISRA 2012 rule 15.4 suggests that "There should be no more than one break or goto statement used to terminate any iteration statement". This rule is of category advisory. Can we consider to use one break in the loop only?

Reply:
Updated the code to remove the use of break.